ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -605,8 +602,32 @@ const DataSetPage = fp.flow(withUserProfile(), withCurrentWorkspace(), withUrlPa
       this.setState({selectedDomainValuePairs: valuesSelected, dataSetTouched: true});
     }
 
+    // Add the values to the valueSet as well as the selectedDomain list so that they are
+    // all selected by default
+    updateValueSets(updatedValueSets, newValueSets) {
+      this.setState({
+        valueSets: updatedValueSets.concat(newValueSets),
+        valuesLoading: false
+      });
+      const allValuesSelected = this.state.selectedDomainValuePairs;
```

Comment:
(opt / nitpick) The name choice here feels a little off. Is this variable meant to hold "all currently-selected values", or is it mean to indicate that "all values should be selected"?

I think, in terms of communicating the intent to readers, a more generic choice would be to simply destructure the state variable and include a comment:

// Newly-fetched values should be selected by default, so we add them to the selection-tracking state property.
const {selectedDomainValuePairs} = this.state;
...
this.setState({selectedDomainValuePairs: selectedDomainValuePairs});